pkg/docker/dockerimage/dockerimage.go

```diff
@@ -1133,23 +1134,45 @@ func inspectFile(
 				object.ContentType = ContentTypeUTF8
 				if utf8Detector.Dump {
 					if utf8Detector.Archive != nil {
-						fileInfo := &utf8FileInfo{
-							name:    hash,
-							size:    object.Size,
-							modtime: object.ModTime,
-						}
-						header, err := tar.FileInfoHeader(fileInfo, hash)
-						if err != nil {
-							return err
-						}
-						header.Name = hash
-						err = utf8Detector.Archive.Writer.WriteHeader(header)
-						if err != nil {
-							return err
-						}
-						_, err = utf8Detector.Archive.Writer.Write(data)
-						if err != nil {
-							return err
+						if utf8Detector.MaxSizeBytes != 0 && object.Size > int64(utf8Detector.MaxSizeBytes) {
+							data = []byte(fmt.Sprintf("hash: %s size: %d [content omitted because too large]", hash, object.Size))
```

Comment:
Why is the file information included in the archive if it's too big?

Reply:
this is a effectively a tombstone. the output is guaranteed to contain an entry for every utf8 decodable file in the container. it also provides a minimum of information about the file.

Reply:
is this how you track if a file is utf8 encoded by checking if the file is in this archive?

Reply:
it's more of an invariant. a file is either meant to be included in this tar'd content, or not. an unexpected absence would be an error. in this case, i want to omit the body of content because it's too large to store. i still want to maintain the invariants of the data as a whole.

Reply:
my use case is, given a that a file is utf8 encoded, i may want to look up it's contents. i know if it's utf8 encoded by looking at the slim report output. when i look it up, i expected to find something, even if that is not the actual contents.

Reply:
How about we make it truncated data if it's useful to have something there and we'll know if it's truncated when the size in the report doesn't match the size in the archive?